server.go

```diff
@@ -97,16 +90,44 @@ func (s *Server) ServeHTTPS(addr, keyfile, certfile string, ready *chan bool) er
 	if s.listener, err = listenTLS(addr, keyfile, certfile); err != nil {
 		return err
 	}
+	s.tls = true
+	fmt.Printf("Listen http on %s\n", addr)
+	return s.doServe(ready)
+}
+
+func (s *Server) doServe(ready *chan bool) error {
+	// A dirty trick to associate a connection with the http.Request it
+	// contains. In "net/http/server.go", handler will be called
+	// immediately after ConnState changed to StateActive, so it's safe to
+	// loop through all elements in a channel to find a match remote addr.
+	q := make(chan net.Conn, 10)
```

Comment:
Isn't there a way to do this without limiting the buffer to 10?
The whole thing is very hacky, which is caused by the limitations of the Go library API, but I think we can at least remove this limitation.